### PR TITLE
ci(datasets): Run tensorflow tests separately from other dataset tests

### PR DESCRIPTION
**Pull request by merelcht**
## Description
For some mysterious reason `test_tensorflow_model_dataset.py` fails on Github Actions when run as part of the full test suite. The tests run fine locally as well as on gitpod. I discovered this while reviewing: https://github.com/kedro-org/kedro-plugins/pull/253 

Some of the stacktraces from the failing builds:
```
tests/tensorflow/test_tensorflow_model_dataset.py:449: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/unittest/mock.py:1081: in __call__
    return self._mock_call(*args, **kwargs)
/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/unittest/mock.py:1085: in _mock_call
    return self._execute_mock_call(*args, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <MagicMock spec='str' id='139697796377376'>, args = ()
kwargs = {'filepath': '/tmp/pytest-of-runner/pytest-0/popen-gw0/test_save_and_load_with_device0/test_tf', 'load_args': {'tf_device': '/CPU:0'}, 'version': Version(load=None, save='2023-10-05T15.43.50.383Z')}
effect = <tuple_iterator object at 0x7f0ded87a9d0>

    def _execute_mock_call(self, /, *args, **kwargs):
        # separate from _increment_mock_call so that awaited functions are
        # executed separately from their call, also AsyncMock overrides this method
    
        effect = self.side_effect
        if effect is not None:
            if _is_exception(effect):
                raise effect
            elif not _callable(effect):
>               result = next(effect)
E               StopIteration

/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/unittest/mock.py:1142: StopIteration
```

```
FAILED tests/tensorflow/test_tensorflow_model_dataset.py::test_deprecation[TensorFlowModelDataSet-kedro_datasets.tensorflow.tensorflow_model_dataset] - AttributeError: Mock object has no attribute '__name__'
FAILED tests/tensorflow/test_tensorflow_model_dataset.py::TestTensorFlowModelDataset::test_save_and_load[None-None-None] - AssertionError: 
Not equal to tolerance rtol=1e-06, atol=1e-06

(shapes (2, 2, 1), (0,) mismatch)
 x: array([[[ 0.005867],
        [ 0.005867]],
...
 y: array([], dtype=float64)
```
## Development notes
I separated the tensorflow test runs from the other datasets tests. This has resolved the issue. 

## Checklist

- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the relevant `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

**Review by ankatiyar — APPROVED** [review]
That's so strange 🤔 but this workaround looks good to me!

File: .github/workflows/unit-tests.yml [hunk 1]
```diff
@@ -49,9 +49,12 @@ jobs:
...
       - name: pip freeze
         run: pip freeze
-      - name: Run unit tests for Linux / all plugins
-        if: inputs.os != 'windows-latest'
+      - name: Run unit tests for Linux / kedro-airflow, kedro-docker, kedro-telemetry
+        if: inputs.os != 'windows-latest' && inputs.plugin != 'kedro-datasets'
         run: make plugin=${{ inputs.plugin }} test
+      - name: Run unit tests for Linux / kedro-datasets
+        if: inputs.os != 'windows-latest' && inputs.plugin == 'kedro-datasets'
+        run: make dataset-tests
```
Comment (SajidAlamQB) [inline .github/workflows/unit-tests.yml:57]: Maybe we should leave a comment why this was split.

**Review by SajidAlamQB — APPROVED** [review]
I think it needs further investigation, but I am happy for this to go in as a temporary fix, maybe a follow up issue can be made?

**Comment by merelcht** [comment]
> I think it needs further investigation, but I am happy for this to go in as a temporary fix, maybe a follow up issue should be made.

I created https://github.com/kedro-org/kedro-plugins/issues/367 a couple of days ago. Will update it to also mention the issues that required this specific work-around.

**Pull request merged**